### PR TITLE
improve metashape data import

### PR DESCRIPTION
**Pull request by f-dy**
- support multiple sensors
- support fisheye and equirectangular sensors
- support multiple components with transforms
- fix bug when image name has a '.' in it, excluding the prefix separator

### CODE REVIEW

File: nerfstudio/process_data/metashape_utils.py [hunk 1]
```diff
@@ -62,68 +62,108 @@ def metashape_to_json(  # pylint: disable=too-many-statements
...
-    data["camera_model"] = CAMERA_MODELS["perspective"].value
+    if sensor_type[0] == "frame":
+        data["camera_model"] = CAMERA_MODELS["perspective"].value
+    elif sensor_type[0] == "fisheye":
+        data["camera_model"] = CAMERA_MODELS["fisheye"].value
+    elif sensor_type[0] == "spherical":
+        data["camera_model"] = CAMERA_MODELS["equirectangular"].value
+    else:
+        # Cylindrical and RPC sensor types are not supported
+        raise ValueError("Unsupported Metashape sensor type '{sensor_type[0]}'")
```
Comment (tancik) [inline nerfstudio/process_data/metashape_utils.py:79]: Missing `f` for f-string

**Review by tancik — COMMENTED** [review]
Thanks for improving metashape support!

**Review by tancik — APPROVED** [review]
LGTM

**Comment by gad-ass** [comment]
Hey, is the fisheye camera importing already working?
Because for me something is wrong and I'm not sure if I'm doing something wrong, or it's not working yet. :)

https://github.com/nerfstudio-project/nerfstudio/assets/32915902/cbca50b1-0947-4843-b07f-971b4f00695d

**Comment by devernay** [comment]
Yes it's working well, but you want to also use masks in your case (photo not covering the whole image)

**Comment by gad-ass** [comment]
I had to change in transforms.json camera model to "OPENCV_FISHEYE" :)
ps. can you give me a hint how can I apply masks for fisheye?
I tried to export png's with alpha channel, but It looks like it's ignoring them :)
Thanks!

**Comment by f-dy** [comment]
computing the mask:
```
cat > compute_mask.py <<EOF
#!/usr/bin/env python
import cv2
import sys
import numpy as np

iterations = 8

# Import all image files with the .jpg extension
files = sys.argv[1:]
kernel = cv2.getStructuringElement(cv2.MORPH_RECT, (3, 3))
for i, f in enumerate(files):
    img = cv2.imread(f, cv2.IMREAD_GRAYSCALE)
    img = cv2.morphologyEx(img, cv2.MORPH_CLOSE, kernel, iterations=iterations)
    if i == 0:
        avg = img
        # max = img
    else:
        alpha = 1.0 / (i + 1)
        beta = 1.0 - alpha
        avg = cv2.addWeighted(img, alpha, avg, beta, 0.0)

avg = cv2.morphologyEx(avg, cv2.MORPH_CLOSE, kernel, iterations=iterations)
avg = np.uint8(avg)
ret, avg = cv2.threshold(avg, 0, 255, cv2.THRESH_OTSU)
cv2.imwrite("mask.png", avg)
print("wrote mask.png")
EOF
```
adding the mask to transforms.json:
```
cat > add_mask_to_transforms_json.py <<EOF
#!/usr/bin/env python
import sys
import json

if len(sys.argv) != 3:
    print(f"Usage: {sys.argv[0]} input_transforms.json output_transforms.json")
    sys.exit(1)
with open(sys.argv[1]) as input_file:
    file_contents = input_file.read()
parsed_json = json.loads(file_contents)
for frame in parsed_json["frames"]:
    frame["mask_path"] = "masks/mask.png"
with open(sys.argv[2], "w") as output_file:
    json.dump(parsed_json, output_file, indent=4)
EOF
```
downsizing the masks:
```
cat > downsize_mask.py <<EOF
#!/usr/bin/env python
import cv2
import sys
import json
from pathlib import Path

if len(sys.argv) != 2:
    print(f"Usage: {sys.argv[0]} path_to/mask.png")
    print(f"Output is path_to/masks_{downscale}/mask.png")
    sys.exit(1)
mask_path = Path(sys.argv[1])
mask = cv2.imread(str(mask_path), cv2.IMREAD_GRAYSCALE)
height, width = mask.shape[:2]
processed_data_dir = mask_path.parent
downscale_factors = [2, 4, 8]
for downscale in downscale_factors:
    mask_path_i = processed_data_dir / f"masks_{downscale}"
    mask_path_i.mkdir(exist_ok=True)
    mask_path_i = mask_path_i / "mask.png"
    mask_i = cv2.resize(
        mask, (width // downscale, height // downscale), interpolation=cv2.INTER_NEAREST
    )
    cv2.imwrite(str(mask_path_i), mask_i)
    print(f"Wrote {mask_path_i}")
EOF
```

**Pull request merged**